### PR TITLE
feat: promote js_library to public API

### LINKED ISSUE #149: promote the js_library rule to the public API
I looked through the definitions file and couldn't find any `nodejs_library` definition, does that means that I can only have exactly one global `BUILD` and have a monolithic building process?

### PR DESCRIPTION
**Pull request by alexeagle**
Fixes #149

### CODE REVIEW

**Comment by alexeagle** [comment]
Thanks to @mistic for doing the work required to make `js_library` work the right way :)

**Review by gregmagolan — COMMENTED** [review]
:rocket:

There are more instances of
```
load("@build_bazel_rules_nodejs//internal/js_library:js_library.bzl", "js_library")
```

that can be changed.

File: internal/js_library/js_library.bzl [hunk 1]
```diff
@@ -37,6 +37,41 @@ load(
...
+    "amd_names": attr.string_dict(
+        doc = _AMD_NAMES_DOC,
+    ),
+    "deps": attr.label_list(
+        doc = """Transitive dependencies of the package.
+        It should include fine grained npm dependencies from the sources
+        or other targets we want to include in the library but also propagate their own deps.""",
+    ),
+    "is_windows": attr.bool(
+        doc = "Automatically set by macro",
```
Comment (gregmagolan) [inline internal/js_library/js_library.bzl:50]: "Internal use only. Automatically set by macro"?

File: internal/js_library/js_library.bzl [hunk 2]
```diff
@@ -37,6 +37,41 @@ load(
 _AMD_NAMES_DOC = """Mapping from require module names to global variables.
 This allows devmode JS sources to load unnamed UMD bundles from third-party libraries."""
 
+_ATTRS = {
+    "amd_names": attr.string_dict(
+        doc = _AMD_NAMES_DOC,
+    ),
+    "deps": attr.label_list(
+        doc = """Transitive dependencies of the package.
```
Comment (gregmagolan) [inline internal/js_library/js_library.bzl:45]: remove docstring here and have it documented in macro only?

File: internal/js_library/js_library.bzl [hunk 3]
```diff
@@ -37,6 +37,41 @@ load(
...
+        doc = """A subset of srcs that are javascript named-UMD or
+        named-AMD for use in rules such as ts_devserver.
+        They will be copied into the package bin folder if needed.""",
+        allow_files = True,
+    ),
+    "package_name": attr.string(
+        doc = """Optional package_name that this package may be imported as.""",
+    ),
+    "srcs": attr.label_list(
+        doc = """The list of files that comprise the package.
```
Comment (gregmagolan) [inline internal/js_library/js_library.bzl:69]: remove docstring here and have it documented in macro only?

File: internal/js_library/js_library.bzl [hunk 4]
```diff
@@ -37,6 +37,41 @@ load(
 _AMD_NAMES_DOC = """Mapping from require module names to global variables.
 This allows devmode JS sources to load unnamed UMD bundles from third-party libraries."""
 
+_ATTRS = {
+    "amd_names": attr.string_dict(
+        doc = _AMD_NAMES_DOC,
```
Comment (gregmagolan) [inline internal/js_library/js_library.bzl:42]: mark as non-public and pull in string from _AMD_NAMES_DOC?

File: internal/js_library/js_library.bzl [hunk 5]
```diff
@@ -37,6 +37,41 @@ load(
...
+        mandatory = True,
+    ),
+    # module_name for legacy ts_library module_mapping support
+    # which is still being used in a couple of tests
+    # TODO: remove once legacy module_mapping is removed
+    "module_name": attr.string(
+        doc = "Internal use only. It will be removed soon.",
+    ),
+    "named_module_srcs": attr.label_list(
+        doc = """A subset of srcs that are javascript named-UMD or
```
Comment (gregmagolan) [inline internal/js_library/js_library.bzl:60]: indicate it is non-public and may be removed in the future?

**Review by gregmagolan — APPROVED** [review]
🐰

**Pull request merged**